app/packages/state/src/utils.ts

```diff
@@ -118,11 +119,11 @@ export type ResponseFrom<TQuery extends { response: unknown }> =
 export const getSavedViewName = (context: RoutingContext<any>): string => {
   const datasetName = getDatasetName(context);
   if (datasetName) {
-    const queryString = window.location.search;
+    const queryString = context.history.location.search;
     const params = new URLSearchParams(queryString);
     const viewName = params.get("view");
     if (viewName) {
-      return viewName;
+      return decodeURIComponent(viewName);
```

Comment:
is `getSavedViewName` used where the name specifically (not the slug) is required? If so, then this might not work. For example if `viewName = "saved view"`, then `slug = "saved-view"`, and`decodeURIComponent(viewName)==slug` but `decodeURIComponent(viewName)!=viewName`

Reply:
This method was always assuming the slug, but I will rename.